apps/frontend/src/pages/profile/profile.tsx

```diff
@@ -73,6 +74,24 @@ const Profile: React.FC = () => {
 
 	const isLoading = dataStatus === DataStatus.PENDING;
 
+	const [isDirty, setIsDirty] = useState<boolean>(false);
+
+	const blocker = useBlocker(({ currentLocation, nextLocation }) => {
+		return isDirty && currentLocation.pathname !== nextLocation.pathname;
+	});
+
+	const handleCancelPopupClick = useCallback((): void => {
+		if (blocker.state === "blocked") {
+			blocker.reset();
+		}
+	}, [blocker]);
+
+	const handleConfirmPopupClick = useCallback((): void => {
+		if (blocker.state === "blocked") {
+			blocker.proceed();
+		}
+	}, [blocker]);
```

Comment:
I feel like this will be duplicated lots of times, let's create a separate hook for that logic